geoips/pydantic/bases.py

```
@@ -0,0 +1,458 @@
+    """
+
+    interface: PythonIdentifier = Field(
+        ...,
+        description="""Name of the plugin's interface. Run geoips list interfaces to see
+        available options.""",
+    )
+    family: PythonIdentifier = Field(..., description="Family of the plugin.")
+    name: PythonIdentifier = Field(..., description="Plugin name.")
+    docstring: str = Field(..., description="Docstring for the plugin in numpy format.")
+    description: str = Field(
+        None,
+        description=("A short description or defaults to first line from docstring."),
+    )
+    package: PythonIdentifier = (
+        Field(..., description="Package that contains this plugin."),
+    )
+    relpath: str = Field(
+        None, description="Path to the plugin file relative to its parent package."
+    )
+    abspath: str = Field(None, description="Absolute path to the plugin file.")
+
+    @model_validator(mode="before")
+    def _derive_package_name(
+        cls: type["PluginModel"], values: dict[str, str | int | float | None]
+    ):
+        """
+        'package' value is derived by calling ``get_plugin_metadata()``.
+
+        Parameters
+        ----------
+        values : dict
+            Dictionary of field values before validation.
+
+        Returns
+        -------
+        dict
+            Updated dictionary with the `package` field set based on the
+            metadata retrived from ``get_plugin_metadata()``.
+        """
+        # name is guaranteed to exist due to Pydantic validation.
+        # No need to raise an error for 'name'.
+        metadata = workflows.get_plugin_metadata(values.get("name"))
+        if "package" not in metadata:
+            err_msg = (
+                "Metadata for '%s' workflow plugin must contain 'package' key."
+                % values.get("name")
+            )
+            LOG.critical(err_msg)
+            raise ValueError(err_msg)
+        values["package"] = metadata.get("package")
```

I think that, rather than having this `if` statement, we should make this like this:

```
try:
    metadata = workflows.get_plugin_metadata(values.get("name"))
    values["package"] = metadata.get("package")
except KeyError:  # Later this will be PluginRegistryError
    values["package"] = None
return values

---

Actually, I think this is wrong. In the case where we are not going to generate the plugin registry, we need to raise an error. We may need to add some state information to the plugin registry to help us here.

---

new issue for path validation : https://github.com/NRLMMD-GEOIPS/geoips/issues/904

---

I have created an [issue](https://github.com/NRLMMD-GEOIPS/geoips/issues/906) to enhance the above exception handling using state information from the plugin registry.

This issue is in addition to the path validation issue which deals with the state information and other bugs.